loader/src/packfile_io.rs

```diff
@@ -52,20 +66,69 @@ impl Drop for PackfileMessageReader {
         }
     }
 }
+
+struct PackfileMessageReaderBuffer {
+    data: &'static [u8],
+    message_reader: ManuallyDrop<ThreadLocal<capnp::message::Reader<SliceSegments<'static>>>>,
+}
+
+impl PackfileMessageReaderBuffer {
+    pub fn new(data: &'static [u8]) -> std::io::Result<Self> {
+        Ok(PackfileMessageReaderBuffer {
+            data,
+            message_reader: ManuallyDrop::new(ThreadLocal::new()),
+        })
+    }
+}
+
+impl PackfileMessageReader for PackfileMessageReaderBuffer {
+    fn get_reader(&self) -> capnp::Result<pack_file::Reader<'_>> {
+        let message_reader = self.message_reader.get_or_try(|| {
+            // We ensure that the reader is dropped before the mmap so it's ok to cast to 'static here
```

Comment:
Commented out code?

Reply:
Looks like this block happens above at line 50 now.

Reply:
This is cleaned up in the other PR